src/Datadog.Trace.ClrProfiler.Managed/Integrations/Testing/NUnitIntegration.cs

```diff
@@ -0,0 +1,358 @@
+                testMethodCommandType.FullName != NUnitSkipCommandType)
+            {
+                return execute(testMethodCommand, testExecutionContext);
+            }
+
+            Scope scope = CreateScope(testExecutionContext, testMethodCommandType);
+            if (scope is null)
+            {
+                return execute(testMethodCommand, testExecutionContext);
+            }
+
+            using (scope)
+            {
+                object result = null;
+                Exception exception = null;
+                try
+                {
+                    scope.Span.ResetStartTime();
+                    result = execute(testMethodCommand, testExecutionContext);
+                }
+                catch (Exception ex)
+                {
+                    exception = ex;
+                    throw;
+                }
+                finally
+                {
+                    FinishScope(scope, testExecutionContext, exception);
+                }
+
+                return result;
+            }
+        }
+
+        private static Scope CreateScope(object testExecutionContext, Type testMethodCommandType)
+        {
+            Scope scope = null;
+
+            try
+            {
+                if (testExecutionContext.TryGetPropertyValue<object>("TestObject", out object testObject) &&
+                    testExecutionContext.TryGetPropertyValue<object>("CurrentTest", out object currentTest))
+                {
+                    MethodInfo testMethod = null;
+                    object[] testMethodArguments = null;
+                    object properties = null;
+
+                    if (currentTest != null)
+                    {
+                        if (currentTest.TryGetPropertyValue<object>("Method", out object method))
+                        {
+                            method?.TryGetPropertyValue<MethodInfo>("MethodInfo", out testMethod);
+                        }
+
+                        currentTest.TryGetPropertyValue<object[]>("Arguments", out testMethodArguments);
+                        currentTest.TryGetPropertyValue<object>("Properties", out properties);
+                    }
+
+                    if (testMethod != null)
+                    {
+                        string testFramework = "NUnit " + testMethodCommandType.Assembly.GetName().Version;
+                        string testSuite = testMethod.DeclaringType?.FullName;
+                        string testName = testMethod.Name;
+                        string skipReason = null;
+                        List<KeyValuePair<string, string>> testArguments = null;
+                        List<KeyValuePair<string, string>> testTraits = null;
+
+                        // Get test parameters
+                        ParameterInfo[] methodParameters = testMethod.GetParameters();
+                        if (methodParameters?.Length > 0 && testMethodArguments != null)
```

Comment:
Can we check for `testMethodArguments != null` inside the if/else statement (inside the for loop)? It seems like it would still be useful to go down the else branch even if it turns out `testMethodArguments` is null

Reply:
fixed in: https://github.com/DataDog/dd-trace-dotnet/pull/870/commits/af50684975ef32b3416e4f8567fc0eb48ec5e841